DigitalLearningSolutions.Web/Controllers/Signposting/SignpostingSsoController.cs

```diff
@@ -47,5 +54,36 @@ public IActionResult LinkLearningHubSso([FromQuery] LinkLearningHubRequest linkL
             var model = new LinkLearningHubViewModel(isAccountAlreadyLinked, learningHubResourcedId);
             return View(model);
         }
+
+        [HttpGet("ViewResource/{resourceReferenceId}")]
+        public IActionResult ViewResource(int resourceReferenceId)
+        {
+            var delegateId = User.GetCandidateIdKnownNotNull();
+            var learningHubAuthId = userService.GetDelegateUserLearningHubAuthId(delegateId);
+
+            if (!learningHubAuthId.HasValue)
+            {
+                var sessionLinkingId = Guid.NewGuid().ToString();
+                HttpContext.Session.SetString(LinkLearningHubRequest.SessionIdentifierKey, sessionLinkingId);
+
+                var linkingUrl = learningHubLinkService.GetLinkingUrlForResource(resourceReferenceId, sessionLinkingId);
+                return Redirect(linkingUrl);
+            }
+
+            var resourceUrl =
+                learningResourceReferenceService.GetLearningHubResourceLinkByResourceRefId(resourceReferenceId);

```

Comment:
I've been thinking about this, and I think it's a bit more confusing than it is useful to be doing things this way. Let's use the LearningHubResourceService to get this link instead - that will query the API and use whatever it returns, or use data from our db if it can't reach the API - then we're treating resource links the same way we're treating all the other resource details.

Reply:
Sorry to change the spec on you, it should only be a few lines to change and some code to remove.